### ISSUE TITLE
🐛 Biome encountered an unexpected error, probably due to Arabic chars.

### ISSUE THREAD

**Issue by Osama-Elshimy** [biomejs/biome#4231]
### Environment information

```block
CLI:
  Version:                      1.9.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.3.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/9.10.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           true
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

1. Run `biome lint` on a repo where I have Arabic text, I get the following error:

```bash
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_js_factory/src/utils.rs:34:36
Thread Name: biome::worker_7
Message: byte index 1 is not a char boundary; it is inside 'ه' (bytes 0..2) of `هذا الملف غير متوقع `

./src/middlewares/errorHandler.ts internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: byte index 1 is not a char boundary; it is inside 'ه' (bytes 0..2) of `هذا الملف غير متوقع `

  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.


Checked 36 files in 9ms. No fixes applied.
```

I also got these crashing messages from vscode:
- Client Biome: connection to server is erroring. Shutting down server (srouce: biome).
- Request textDocument/codeAction failed (srouce: biome).

### Expected result

`biome` should handle Arabic text and work correctly. 

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Duplicate of https://github.com/biomejs/biome/issues/4181
